pkg/operator/ceph/cluster/controller.go

```diff
@@ -494,23 +496,56 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 	logger.Infof("update event for cluster %s is supported, orchestrating update now", newClust.Namespace)
 
 	// if the image changed, we need to detect the new image version
+	versionChanged := false
 	if oldClust.Spec.CephVersion.Image != newClust.Spec.CephVersion.Image {
 		logger.Infof("the ceph version changed. detecting the new image version...")
+		versionChanged = true
 		version, err := cluster.detectCephVersion(newClust.Spec.CephVersion.Image, 15*time.Minute)
 		if err != nil {
 			logger.Errorf("unknown ceph major version. %+v", err)
 			return
 		}
 		cluster.Info.CephVersion = *version
 	} else {
-		logger.Infof("ceph version is still %s on image %s", &cluster.Info.CephVersion, cluster.Spec.CephVersion.Image)
+		// At this point, clusterInfo might not be initialized
```

Comment:
This comment is written as a sentence, but the lines are all arbitrary lengths/statements. I think if this is reformatted to be just statements ending in periods, this comment could be one line less in length.